File: FSharp/Tests/HotelBookingsPossibleTests.fs
namespace Tests

module HotelBookingsPossibleTests =
    open Library.Utils
    open Library.HotelBookingsPossible
    open Xunit

    let data = [|
        // False Test
        [| ~~[ 1; 3; 5 ]; ~~[ 2; 6; 8 ]; ~~1; ~~false |]

        // True Test
        [| ~~[ 1; 3; 5 ]; ~~[ 2; 6; 8 ]; ~~2; ~~true |]
    |]

    [<Theory>]
    [<MemberData("data")>]
    let Test (arrivals, departures, rooms, expected) =
        let actual = Check arrivals departures rooms
        Assert.Equal(expected, actual)

File: CSharp/Tests/HotelBookingsPossibleTests.cs
using Xunit;
using Checker = Library.HotelBookingsPossible;

namespace Tests {
    public class HotelBookingsPossibleTests {
        [Fact]
        public void Test() {
            // Arrange
            var arrivals = new[] { 1, 3, 5 };
            var departures = new[] { 2, 6, 8 };

            // Act
            var trueActual = Checker.Check(arrivals, departures, 2);
            var falseActual = Checker.Check(arrivals, departures, 1);

            // Assert
            Assert.True(trueActual);
            Assert.False(falseActual);
        }
    }
}